lib/hardware_interface/base_hardware_interface.dart

```diff
@@ -0,0 +1,15 @@
+import 'dart:async';
+
+import '../database.dart';
+
+///Extend this class to add custom (non-front-end) functionality to the GUI
+class BaseHardwareInterface {
+  var database = Database();
+  late Timer guiUpdateLoopTimer;
+
+  BaseHardwareInterface() {
+    guiUpdateLoopTimer = Timer.periodic(const Duration(milliseconds: 50), runLoopOnce);
```

Comment:
Why does this need to be initialized in the constructor? why not in the class?

Reply:
In addition I'm sorta questioning the need for a timer in the first place. Why not just provide a `addPeriodic` function in case a interface actually _needs_ to do things?